### ISSUE TITLE
Maintain search state across page navigation

### ISSUE THREAD

**Issue by rsimon** [performant-software/itsb#17]
Currently, the search state (=selected authors and time filter) resets when the user changes between __Trajectories__ and __Intersections__ pages. I played with this a bit, and the reason is the following:

- For each page change, the `mapLoader` function is triggered, causing the data files to get re-loaded
- The `ITSBGraphProvider` creates a new `graph` whenever the data changes [here](https://github.com/performant-software/itsb/blob/main/src/components/ITSBStore/ITSBGraphProvider.jsx#L8-L16).
- Because the graph has changed, the `ITSBSearchHandler` will reset the search [here](https://github.com/performant-software/itsb/blob/main/src/components/ITSBStore/ITSBSearchHandler.jsx#L35-L38).

A quick workaround would be to force the ITSBGraphProvider to __only ever create the graph once, even if the data changes.__ E.g.:

```js
useEffect(() => {
  const { authors, places, itineraries } = props;

  // was: if (authors && places && itineraries) { ...
  if (!graph) {
    const g = new ITSBGraph();
    g.init(authors, places, itineraries);
    setGraph(g);
  }
}, [props.authors, props.places, props.itineraries]);
```

I tested this, and it will retain state across page nav. 

Caution: this would mean that the app won't function properly if a __different__ dataset is loaded after the graph was created.This is never the case in our application, however. A cleaner solution, ultimately, would be to avoid re-triggering the mapLoader somehow.

**Comment by rsimon** [comment]
I just realized there's another issue: since the Store is currently instantiated at the level of the MapPage, it's unavailable to the search page.

This means we'll need to move it up one level, anyway. Probably into Root.jsx. Something like this?

```jsx
export function Root() {
  return (
    <Peripleo>
      <Header />
      <ITSBStore
        authors={authors.itemListElement}
        places={places.features}
        itineraries={itineraries.first.items}
      >
        <Outlet />
      </ITSBStore>
    </Peripleo>
  );
}
```

Now all outlet pages will have access to the graph and search scope - and we can ensure that `authors`, `places` and `itineraries` only get loaded once at the root level.

**Comment by blms** [comment]
Closed by #17
